### ISSUE TITLE
Validator Nodes Must Be Able to Run Archival Sync

### ISSUE THREAD

**Issue by xcthulhu** [casper-network/casper-node#2083]
The current implementation for archival syncing in #1995 can create a long delay for a bonded validator joining the network.

We must support bonded validators running archival nodes.

**Comment by xcthulhu** [comment]
One suggestion put forward by @goral09 is to launch a task in the participating reactor.


One way of doing this is to extract the following from the `run_fast_sync_task`, delete it there and launch it as an effect in the participating reactor:

```rust
    // Sync to genesis
    let mut walkback_block = *fetch_and_store_block_by_hash(effect_builder, trusted_block_header.hash()).await?;;
    loop {
        sync_deploys_and_transfers_and_state(effect_builder, &walkback_block).await?;
        if walkback_block.height() == 0 {
            break;
        } else {
            walkback_block = *fetch_and_store_block_by_hash(
                effect_builder,
                *walkback_block.header().parent_hash(),
            )
            .await?;
        }
    }
```

**Comment by Fraser999** [comment]
The current archival strategy does not require fast-syncing before the current activation point, so closing this ticket for now.

**Comment by xcthulhu** [comment]
@Fraser999 as discussed the legacy archival strategy does not properly validate finality signatures.

**Comment by xcthulhu** [comment]
Closed in favor of the duplicate #2487
